File: fixtures/hub-flagship-scala/src/main/scala/hub/FlagshipRoutes.scala
package hub

import akka.http.scaladsl.model.StatusCodes
import akka.http.scaladsl.server.Directives._
import akka.http.scaladsl.server.Route

/**
 * hub-flagship-scala — 20-route Akka HTTP mirror of hub-flagship-express / kotlin / java.
 * No invented product UI (D6447). Bodies use complete / Map / StatusCodes idioms the hub
 * Scala→WebIR lift understands (brace-bounded routes + path/query refs).
 */
object FlagshipRoutes {
  val routes: Route = concat(
    get(path("/health")) { complete(true) },
    get(path("/ping")) { complete(42) },
    get(path("/version")) { complete(1) },
    get(path("/ready")) { complete("ok") },
    get(path("/count")) { complete(3) },
    get(path("/flag")) { complete("chrysalis") },
    get(path("/build")) { complete(2026) },
    get(path("/tier")) { complete("gold") },
    get(path("/meta")) { complete(Map("service" -> "hub-flagship-scala", "version" -> 1)) },
    post(path("/echo")) { complete(Map("echo" -> true)) },
    get(path("/items")) { complete(true) },
    get(path("/items/{id}")) { complete(Map("id" -> id)) },
    post(path("/items")) { complete(StatusCodes.Created, Map("created" -> true)) },
    get(path("/search")) {
      val q = parameter("q").getOrElse("")
      complete(Map("q" -> q))
    },
    put(path("/items/{id}")) { complete(Map("updated" -> true, "id" -> id)) },
    delete(path("/items/{id}")) { complete(true) },
    patch(path("/items/{id}")) { complete(Map("patched" -> true, "id" -> id)) },
    get(path("/users/{userId}")) { complete(userId) },
    get(path("/stats")) { complete(3) },
    post(path("/notify")) { complete(StatusCodes.Accepted, Map("ok" -> true)) },
  )
}

File: fixtures/hub-flagship-swift/Sources/App/FlagshipRoutes.swift
import Vapor

/**
 * hub-flagship-swift — 20-route Vapor mirror of hub-flagship-express / kotlin / java.
 * No invented product UI (D6447). Bodies use dict / encodeResponse / path-query idioms
 * the hub Swift→WebIR lift understands (brace-bounded closures).
 * Path registration uses multi-segment PathComponents (`"items", ":id"`) where idiomatic.
 */
func flagshipRoutes(_ app: Application) throws {
    app.get("health") { _ in
        return true
    }
    app.get("ping") { _ in
        return 42
    }
    app.get("version") { _ in
        return 1
    }
    app.get("ready") { _ in
        return "ok"
    }
    app.get("count") { _ in
        return 3
    }
    app.get("flag") { _ in
        return "chrysalis"
    }
    app.get("build") { _ in
        return 2026
    }
    app.get("tier") { _ in
        return "gold"
    }
    app.get("meta") { _ in
        return ["service": "hub-flagship-swift", "version": 1]
    }
    app.post("echo") { _ in
        return ["echo": true]
    }
    app.get("items") { _ in
        return true
    }
    app.get("items", ":id") { req in
        let id = req.parameters.get("id")!
        return ["id": id]
    }
    app.post("items") { req in
        return try await ["created": true].encodeResponse(status: .created, for: req)
    }
    app.get("search") { req in
        let q = req.query["q"] ?? ""
        return ["q": q]
    }
    app.put("items", ":id") { req in
        let id = req.parameters.get("id")!
        return ["updated": true, "id": id]
    }
    app.delete("items", ":id") { _ in
        return true
    }
    app.patch("items", ":id") { req in
        let id = req.parameters.get("id")!
        return ["patched": true, "id": id]
    }
    app.get("users", ":userId") { req in
        let userId = req.parameters.get("userId")!
        return userId
    }
    app.get("stats") { _ in
        return 3
    }
    app.post("notify") { req in
        return try await ["ok": true].encodeResponse(status: .accepted, for: req)
    }
}
